Modix.Data/ExpandableQueries/ExpandableQueryProvider.cs

```diff
@@ -25,15 +25,19 @@ public IQueryable CreateQuery(Expression expression)
             if (expression is null)
                 throw new ArgumentNullException(nameof(expression));
 
-            var elementType = expression.Type.GetElementType();
+            var elementType = expression.Type.GetElementType()!;
 
             try
             {
-                return Activator.CreateInstance(typeof(ExpandableQuery<>).MakeGenericType(elementType), new object[] { this, expression }) as IQueryable;
+                return (IQueryable)Activator.CreateInstance(typeof(ExpandableQuery<>).MakeGenericType(elementType), new object[] { this, expression })!;
             }
             catch (TargetInvocationException ex)
             {
-                throw ex.InnerException;
+                if (ex.InnerException is object)
```

Comment:
The constructors are annotated: https://source.dot.net/#System.Private.CoreLib/TargetInvocationException.cs. There doesn't appear to be any usages with `null` in the framework code, but that doesn't mean it can't happen.